src/components/PInput.vue

```diff
@@ -0,0 +1,55 @@
+<template>
+  <input v-on="$listeners"
+    :disabled="disabled"
+    :class="classNames"
+    :type="type"
+    v-bind:value="value"
```

Comment:
You need to make a model property to make this work properly. Add:

```js
  data () {
    return {
      model: this.value
    }
  },
```

Then here change this line to

```js
v-model="model"
@input="$emit('input', $event.target.value)"
```

Also there's an issue that'd make two input listeners so update `$listeners` to `listeners`. Add:

```js
computed: {
    classNames () {
      return [
        'puto-input',
        (this.disabled && '-disabled'),
        (this.loading && '-loading')
      ]
    },

    listeners() {
      // Delete duplicate listeners
      const listeners = {...this.$listeners}
      delete listeners.input
      return listeners
    }
  }

```